src/silverlabnwb/timings.py

```diff
@@ -0,0 +1,101 @@
+        self._format_pixel_time_offsets()
+
+    def _read_relative_times_file(self, file_path):
+        raw_data = pd.read_csv(file_path, names=('RelativeTime', 'CycleTime'),
+                               sep='\t', dtype=np.float64) / 1e6  # convert to seconds
+        self.pixel_time_offsets = raw_data['RelativeTime']
+        self.cycle_time = raw_data['CycleTime'][0]
+
+    def _format_pixel_time_offsets(self):
+        row_increments = np.arange(self.n_pixels_per_line)*self.dwell_time
+        pixel_time_offsets_by_roi = {}
+        for roi_index in np.arange(self.n_rois):
+            start_index = self.n_lines_per_roi * roi_index
+            end_index = start_index + self.n_lines_per_roi
+            row_offsets = self.pixel_time_offsets[start_index:end_index]
+            pixel_time_offsets_by_roi[roi_index] = row_offsets[:, np.newaxis]+row_increments
+        self.pixel_time_offsets = pixel_time_offsets_by_roi
+
+
+class LabViewTimings231(LabViewTimings):
+
+    def __init__(self, relative_times_path, roi_path, dwell_time, n_cycles_per_trial, n_trials):
+        super().__init__(relative_times_path, roi_path, dwell_time)
+        self._format_pixel_time_offsets(n_cycles_per_trial, n_trials)
+
+    def _read_relative_times_file(self, file_path):
+        raw_data = pd.read_csv(file_path, sep='\t', dtype=np.float64) / 1e6
+        self.pixel_time_offsets = raw_data['Image Time [us]']
+        self.pixel_time_offsets = self.pixel_time_offsets[self.pixel_time_offsets != 0]
+
+    def _format_pixel_time_offsets(self, n_cycles_per_trial, n_trials):
+        pixel_time_offsets_by_roi = {}
+        n_lines_per_cycle = self.n_rois * self.n_lines_per_roi
+        row_increments = np.arange(self.n_pixels_per_line) * self.dwell_time
+        for roi_index in np.arange(0, self.n_rois):
+            pixel_time_offsets_by_roi[roi_index] = []
+            for cycle_index in np.arange(0, n_cycles_per_trial * n_trials):
+                start_index = self.n_lines_per_roi * roi_index + n_lines_per_cycle * cycle_index
+                end_index = start_index + self.n_lines_per_roi
+                pixel_time_offsets_by_roi[roi_index].append(self.pixel_time_offsets.values[start_index:end_index])
+            pixel_time_offsets_by_roi[roi_index] = np.reshape(pixel_time_offsets_by_roi[roi_index],
+                                                              (n_trials * n_cycles_per_trial, self.n_lines_per_roi))
+            pixel_time_offsets_by_roi[roi_index] = pixel_time_offsets_by_roi[roi_index][:, :, np.newaxis] + row_increments
+
+        # estimate time for one cycle by averaging the time it takes for the first cycle of each trial.
+        # The n_pixels_per_line * pixel_dwell_time contribution of the last line is negligible.
+        first_cycle_times_for_each_trial_by_roi = []
+        for trial_index in list(range(n_trials)):
+            first_cycle_times_for_each_trial_by_roi.append(pixel_time_offsets_by_roi[self.n_rois-1]
```

Comment:
This looks like it's for the last ROI rather than for each of them? (which makes sense, but the variable name is confusing)